CHANGELOG.md

```diff
@@ -22,6 +22,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [Query Insights] Add exporter support for top n queries ([#12982](https://github.com/opensearch-project/OpenSearch/pull/12982))
 - [Query Insights] Add X-Opaque-Id to search request metadata for top n queries ([#13374](https://github.com/opensearch-project/OpenSearch/pull/13374))
 - Add support for query level resource usage tracking ([#13172](https://github.com/opensearch-project/OpenSearch/pull/13172))
+- [Streaming Indexing] Enhance RestAction with request / response streaming support ([#13772](https://github.com/opensearch-project/OpenSearch/pull/13772))
```

Comment:
Kind of burying the lede here because this entry/commit message/PR title all suggest some low level foundational changes, but this is actually introducing a new `_bulk/stream` user facing API.

On that front, are we sure we've got the API signature correct and that we want to expose this in the next 2.x release without putting it behind a feature flag?

Reply:
> On that front, are we sure we've got the API signature correct and that we want to expose this in the next 2.x release without putting it behind a feature flag?

Those are great questions, let me split it:
 - we've got the API signature correct: no, the API will change 100%, could be experimental indeed
 - that we want to expose this in the next 2.x release without putting it behind a feature flag: I could do that, I am not so worried about it because these actions return with the default transport:
  ```
  $ curl -X POST 'http://localhost:9200/_bulk/stream?pretty' -H "Content-Type: application/json"  -d '                              
  {"update":{"_index":"test1","_id":"1"}}
  {"upsert":{"x":3,"y":5},"script":{"source":"ctx._source.x += 1"}}
  '
  {
    "error" : {
      "root_cause" : [
        {
          "type" : "illegal_state_exception",
          "reason" : "The engine does not support HTTP streaming, unable to serve uri [/_bulk/stream?pretty] and method [POST]"
        }
      ],
      "type" : "illegal_state_exception",
      "reason" : "The engine does not support HTTP streaming, unable to serve uri [/_bulk/stream?pretty] and method [POST]"
    },
    "status" : 500
  }
  ```

The `netty4-reactor` transport is needed and it is already experimental, so I think we should be fine here.

Reply:
To elaborate a bit on that, I was thinking to provide something like `HttpServerTransportCapabilities` to `ActionPlugin` fe to let the REST layer be aware of:
 - is HTTP streaming supported?
 - is HTTP/{1,2,3} supported?

It is proven to be difficult without blowing the changeset :-) But we could work towards that if it makes sense in general.

Reply:
> The netty4-reactor transport is needed and it is already experimental

That seems reasonable so I don't think we need another gate in front of the new API. Thanks!